File: iosApp/iosApp/Presentation/External/ExternalLinkViewModel.swift
//
//  ExternalLinkViewModel.swift
//  ConCafe
//
//  Created by 홍희표 on 2026/03/14.
//

import Foundation
import Combine

@MainActor
final class ExternalLinkViewModel: ObservableObject {
    @Published private(set) var uiState: ExternalLinkUiState

    let event = PassthroughSubject<ExternalLinkEvent, Never>()

    init(title: String, url: String) {
        self.uiState = ExternalLinkUiState(title: title, url: url)
    }

    func onAction(_ action: ExternalLinkAction) {
        switch action {
        case .tapBack:
            event.send(.navigateBack)
        }
    }
}

File: composeApp/src/commonMain/kotlin/com/hhp227/concafe/presentation/external/ExternalLinkViewModel.kt
package com.hhp227.concafe.presentation.external

import androidx.lifecycle.ViewModel
import kotlinx.coroutines.flow.MutableSharedFlow
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.asSharedFlow
import kotlinx.coroutines.flow.asStateFlow

class ExternalLinkViewModel(
    title: String,
    url: String
) : ViewModel() {
    private val _uiState = MutableStateFlow(
        ExternalLinkUiState(
            title = title,
            url = url
        )
    )
    val uiState = _uiState.asStateFlow()

    private val _event = MutableSharedFlow<ExternalLinkEvent>(extraBufferCapacity = 1)
    val event = _event.asSharedFlow()

    fun onAction(action: ExternalLinkAction) {
        when (action) {
            ExternalLinkAction.ClickBack -> _event.tryEmit(ExternalLinkEvent.NavigateBack)
        }
    }
}
